Client.jobs.run is not normalizing storage URIs

All `Client.storage` public methods do expand relative storage URIs, but `Client.jobs.run` does not.

The result of `run` is a vague exception:
```
neuromation.api.core.IllegalArgumentError: Unexpected exception AssertionError: . Path with query: /api/v1/jobs. 
```

--- a/CHANGELOG.D/1464.feature
+++ b/CHANGELOG.D/1464.feature
@@ -1,0 +1,1 @@
+API: `Client.jobs.run()` supports now relative storage URIs for volumes.

--- a/neuromation/api/jobs.py
+++ b/neuromation/api/jobs.py
@@ -26,14 +26,13 @@ from dateutil.parser import isoparse
 from multidict import MultiDict
 from yarl import URL
 
-from neuromation.api.abc import (
+from .abc import (
     AbstractDockerImageProgress,
     ImageCommitFinished,
     ImageCommitStarted,
     ImageProgressPush,
     ImageProgressSave,
 )
-
 from .config import Config
 from .core import _Core
 from .images import (
@@ -43,6 +42,7 @@ from .images import (
 )
 from .parser import Parser, Volume
 from .parsing_utils import LocalImage, RemoteImage, _as_repo_str, _is_in_neuro_registry
+from .url_utils import normalize_storage_path_uri
 from .utils import NoPublicConstructor, asynccontextmanager
 
 
@@ -212,7 +212,7 @@ class Jobs(metaclass=NoPublicConstructor):
     ) -> JobDescription:
         url = self._config.api_url / "jobs"
         payload: Dict[str, Any] = {
-            "container": _container_to_api(container),
+            "container": _container_to_api(container, self._config),
             "is_preemptible": is_preemptible,
         }
         if name:
@@ -650,7 +650,7 @@ def _container_from_api(data: Dict[str, Any], parse: Parser) -> Container:
     )
 
 
-def _container_to_api(container: Container) -> Dict[str, Any]:
+def _container_to_api(container: Container, config: Config) -> Dict[str, Any]:
     primitive: Dict[str, Any] = {
         "image": _as_repo_str(container.image),
         "resources": _resources_to_api(container.resources),
@@ -664,7 +664,7 @@ def _container_to_api(container: Container) -> Dict[str, Any]:
     if container.env:
         primitive["env"] = container.env
     if container.volumes:
-        primitive["volumes"] = [_volume_to_api(v) for v in container.volumes]
+        primitive["volumes"] = [_volume_to_api(v, config) for v in container.volumes]
     if container.tty:
         primitive["tty"] = True
     return primitive
@@ -725,9 +725,12 @@ def _job_telemetry_from_api(value: Dict[str, Any]) -> JobTelemetry:
     )
 
 
-def _volume_to_api(volume: Volume) -> Dict[str, Any]:
+def _volume_to_api(volume: Volume, config: Config) -> Dict[str, Any]:
+    uri = normalize_storage_path_uri(
+        volume.storage_uri, config.username, config.cluster_name
+    )
     resp: Dict[str, Any] = {
-        "src_storage_uri": str(volume.storage_uri),
+        "src_storage_uri": str(uri),
         "dst_path": volume.container_path,
         "read_only": bool(volume.read_only),
     }

--- a/tests/api/test_jobs.py
+++ b/tests/api/test_jobs.py
@@ -1016,6 +1016,106 @@ async def test_job_run_no_volumes(
         assert ret == _job_description_from_api(JSON, client.parse)
 
 
+async def test_job_run_with_relative_volume_uris(
+    aiohttp_server: _TestServerFactory, make_client: _MakeClient
+) -> None:
+    JSON = {
+        "id": "job-cf519ed3-9ea5-48f6-a8c5-492b810eb56f",
+        "status": "failed",
+        "history": {
+            "status": "failed",
+            "reason": "Error",
+            "description": "Mounted on Avail\\n/dev/shm     " "64M\\n\\nExit code: 1",
+            "created_at": "2018-09-25T12:28:21.298672+00:00",
+            "started_at": "2018-09-25T12:28:59.759433+00:00",
+            "finished_at": "2018-09-25T12:28:59.759433+00:00",
+        },
+        "owner": "owner",
+        "cluster_name": "default",
+        "uri": "job://default/owner/job-cf519ed3-9ea5-48f6-a8c5-492b810eb56f",
+        "container": {
+            "image": "gcr.io/light-reality-205619/ubuntu:latest",
+            "command": "date",
+            "resources": {
+                "cpu": 1.0,
+                "memory_mb": 16384,
+                "gpu": 1,
+                "shm": False,
+                "gpu_model": "nvidia-tesla-p4",
+            },
+        },
+        "http_url": "http://my_host:8889",
+        "ssh_server": "ssh://my_host.ssh:22",
+        "ssh_auth_server": "ssh://my_host.ssh:22",
+        "is_preemptible": False,
+    }
+
+    async def handler(request: web.Request) -> web.Response:
+        data = await request.json()
+        assert data == {
+            "container": {
+                "image": "submit-image-name",
+                "command": "submit-command",
+                "http": {"port": 8181, "requires_auth": True},
+                "resources": {
+                    "memory_mb": 16384,
+                    "cpu": 7.0,
+                    "shm": True,
+                    "gpu": 1,
+                    "gpu_model": "test-gpu-model",
+                },
+                "volumes": [
+                    {
+                        "src_storage_uri": "storage://default/user/path",
+                        "dst_path": "/container/my_path",
+                        "read_only": False,
+                    },
+                    {
+                        "src_storage_uri": "storage://default/otheruser/path",
+                        "dst_path": "/container/other_user_path",
+                        "read_only": True,
+                    },
+                    {
+                        "src_storage_uri": "storage://othercluster/otheruser/path",
+                        "dst_path": "/container/other_cluster_path",
+                        "read_only": True,
+                    },
+                ],
+            },
+            "is_preemptible": False,
+            "cluster_name": "default",
+        }
+
+        return web.json_response(JSON)
+
+    app = web.Application()
+    app.router.add_post("/jobs", handler)
+
+    srv = await aiohttp_server(app)
+
+    async with make_client(srv.make_url("/")) as client:
+        resources = Resources(16384, 7, 1, "test-gpu-model", True, None, None)
+        volumes: List[Volume] = [
+            Volume(URL("storage:path"), "/container/my_path", False),
+            Volume(URL("storage:/otheruser/path"), "/container/other_user_path", True,),
+            Volume(
+                URL("storage://othercluster/otheruser/path"),
+                "/container/other_cluster_path",
+                True,
+            ),
+        ]
+        container = Container(
+            image=RemoteImage.new_external_image(name="submit-image-name"),
+            command="submit-command",
+            resources=resources,
+            volumes=volumes,
+            http=HTTPPort(8181),
+        )
+        ret = await client.jobs.run(container=container, is_preemptible=False)
+
+        assert ret == _job_description_from_api(JSON, client.parse)
+
+
 async def test_job_run_preemptible(
     aiohttp_server: _TestServerFactory, make_client: _MakeClient
 ) -> None:

